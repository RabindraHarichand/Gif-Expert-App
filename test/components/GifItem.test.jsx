const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe("Testing GifItem component", () => {
  const title = "One Piece";
  const url = {
    url: "https://slfajfl.com",
  };

  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("should show image with url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole("img").src).toBe(url.url + "/");
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url.url + "/");
    expect(alt).toBe(title);
  });

  test("should show title in component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
