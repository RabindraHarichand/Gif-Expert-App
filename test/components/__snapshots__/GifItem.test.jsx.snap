// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="https://slfajfl.com"
    />
    <p>
      One Piece
    </p>
  </div>
</div>
`;
